lib/utils.pm

```diff
@@ -1044,15 +1044,29 @@ sub ensure_serialdev_permissions {
 }
 
 =head2 exec_and_insert_password
-Execute a command that ask for a password and insert it.
+
+    exec_and_insert_password($cmd);
+
+ 1. Execute a command that ask for a password.
+ 2. Detects password prompt
+ 3. Insert password and hits enter
+
 =cut
 sub exec_and_insert_password {
-    my $cmd = shift;
+    my ($cmd) = @_;
+    my $hashed_cmd = hashed_string("SR$cmd");
     type_string "$cmd";
```

Comment:
You should probably wait for the shell prompt here (e.g. `wait_serial(serial_term_prompt(), undef, 0, no_regex => 1);`)

Reply:
of course that only works on serial terminal